### DISCUSSION TITLE
Incorporating go-ethereum updates

### DISCUSSION THREAD

**Discussion by YoshihitoAso** [ConsenSys/quorum#1617] [Q&A] [answered]
GoQuorum is currently based on go-ethereum v1.10.3. Do you plan to incorporate the latest go-ethereum code in the GoQuorum project?

Of course there are changes that are not directly related to GoQuorum, but I believe that we should incorporate changes that will improve GoQuorum.

**Comment by antonydenyer** [comment, marked as answer]
Agreed it's something that we think is important - certainly from a security point of view. As a community we have a couple of options:

1. diverge from geth and update dependencies and cherry-pick code fixes. 
2. stay on track with geth and work through the changes and incorporate them accordingly

The problem is that merging upstream changes from geth is time consuming and generally a thankless task, consequently it competes with other priorities. Suffice to say every time we merge from upstream there's a risk of introducing bugs on things we've previously implemented or fixed (e.g https://github.com/ConsenSys/quorum/pull/1616).

Either way, improvements to https://github.com/ConsenSys/quorum-acceptance-tests are always welcomed as are PRs on https://github.com/ConsenSys/quorum 

If it's something you want to work on, create an issue and 'claim' it so we do not duplicate work

**Reply by YoshihitoAso** [reply]
Thank you for your response.

In my opinion, GoQuorum should "gradually" cherry-pick useful improvements. It is difficult to keep up with a particular version of geth because there are already so many differences.

Our development team can help with those improvements. We will propose issues and submit PRs for any improvements we think should be incorporated.
